src/MemoryCache.ts

```diff
@@ -0,0 +1,67 @@
+import { ICache } from "./types/Cache";
+import { SecondLevelDomain } from "./types/SecondLevelDomain";
+import { TopLevelDomain } from "./types/TopLevelDomain";
+
+export class MemoryCache implements ICache {
+  tld: Map<string, TopLevelDomain>;
+  slds: Map<string, Map<string, SecondLevelDomain>>;
+
+  constructor() {
+    this.tld = new Map<string, TopLevelDomain>();
+    this.slds = new Map<string, Map<string, SecondLevelDomain>>();
+  }
+
+  getTld(
+    tldHash: string
+  ):Promise<TopLevelDomain | undefined> {
+    return Promise.resolve(this.tld.get(tldHash));
+  }
+
+  getTlds(): Promise<TopLevelDomain[]> {
+    return Promise.resolve(Array.from(this.tld.values()) || []);
+  }
+
+  getSld(
+    tldHash: string,
+    sldHash: string
+  ): Promise<SecondLevelDomain | undefined> {
+    if (this.slds.has(tldHash)) {
+      const sldCache = this.slds.get(tldHash);
+      if (sldCache) {
+        return Promise.resolve(sldCache.get(sldHash));
+      }
+    }
+    return Promise.resolve(undefined);
```

Comment:
I believe rejecting instead of resolving undefined might be a good idea here. That being said we can introduce error state to this project so the consumer can handle what is needed

Reply:
Returning a CacheResult object may be useful here.

```ts
interface CacheResult<T> {
  result?: T;
  errorMessage?: string;
}
```

Or

```ts
type CacheResponse<T> = T | { errorMessage: string };
```

Definitely don't want to throw any errors here since missing SLDs will occur often and this is an expected and recoverable "error".